demo/cluster/setup.sh

```
@@ -1147,8 +1386,36 @@ MRAP_EOF
+    # Kyverno only discovers API resources at startup — CRDs registered after Kyverno
+    # starts are invisible to it until it restarts. Crossplane registers ~300 CRDs after
+    # Kyverno is installed, so ClusterProviderConfig is always unknown to Kyverno on the
+    # first attempt. Restart the admission controller to force re-discovery before apply.
+    log_info "Restarting Kyverno admission controller to pick up new CRDs from Crossplane..."
+    kubectl rollout restart deployment/kyverno-admission-controller -n kyverno 2>&1
+    kubectl rollout status deployment/kyverno-admission-controller -n kyverno --timeout=120s 2>&1
+
+    # Apply ClusterProviderConfig for in-cluster identity.
+    local cp_result
+    cp_result=$(kubectl apply -f "${SCRIPT_DIR}/manifests/providerconfig-k8s.yaml" 2>&1)
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)